sdk/file_sdk/Common/ifile_format.h

```diff
@@ -0,0 +1,40 @@
+#ifndef IFILE_FORMAT_H
+#define IFILE_FORMAT_H
+
+#include <string>
+#include <vector>
+#include <IStream.h>
+#include "tag.h"
+
+using std::string;
+using std::vector;
+using rmscrypto::api::IStream;
+
+namespace mip{
+namespace file {
+
+class IFileFormat
+{
+public:
+
+  virtual string GetOriginalExtension() = 0;
+
+  virtual void SetTags(const vector<Tag>& tags) = 0;
+
+  virtual const vector<Tag>& GetTags() = 0;
+
+  virtual void Commit(std::shared_ptr<IStream> file, string& extension) = 0;
+
+protected:
+
+  string mExtension;
+
+  std::shared_ptr<IStream> mFile;
+
+  vector<Tag> mTags;

```

Comment:
moved all to FileFormat

---
In reply to: [134425629](https://github.com/AzureAD/rms-sdk-for-cpp/pull/133#discussion_r134425629) [](ancestors = 134425629)